### PR TITLE
ARGO-418 Mgo Connection handling optimization. Godep refs fixed. 

### PR DESCRIPTION
**Pull request by kaggis**
- [x] Optimize Mongo Session handling
- [x] Add handler for clearing router context
- [x] Add Handler for clearing context
- [x] Fix Topic Not Found case in Topic Show

### CODE REVIEW

**Comment by kaggis** [comment]
please review 
/cc @pkoro @skanct @themiszamani

File: config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "port":8080,
-  "broker_hosts":["localhost:9092"],
+  "broker_hosts":["as01.hadoop:9092"],
```
Comment (pkoro) [inline config.json]: This must have slipped in during your commit. Better change it back to localhost.

File: stores/mongo.go [hunk 2]
```diff
@@ -13,54 +13,60 @@ import (
...
+	mong.Server = server
+	mong.Database = db
 	return &mong
 }
 
 // Close is used to close session
 func (mong *MongoStore) Close() {
-	// mong.Session.Close()
+	mong.Session.Close()
+	log.Printf("Session Closed...")
```
Comment (pkoro) [inline stores/mongo.go]: Maybe it's better to remove this line here.

**Comment by pkoro** [comment]
A have left a couple of minor comments inline. Beyond those maybe it's best to remove also some commented out blocks of code. Other than these +1 from me.

**Pull request merged**